### PR TITLE
[XLIFF] Support for blocks in localized fields and field collections containing localized fields

### PR DESCRIPTION
**Pull request by mcuenez**
Added support for exporting and importing both **blocks in localized fields** and **field collections with localized fields**.

### CODE REVIEW

File: lib/Translation/ExportDataExtractorService/DataExtractor/DataObjectDataExtractor.php [hunk 1]
```diff
@@ -294,6 +330,78 @@ protected function addLocalizedFieldsInBricks(
...
+
+                $getter = 'get' . ucfirst($fieldDefinition->getName());
+
+                /** @var DataObject\Fieldcollection $fieldCollection */
+                if (!$fieldCollection = $object->$getter()) {
+                    continue;
+                }
+
+                $itemFieldDefinitions = $fieldCollection->getItemDefinitions();
+                if (empty($itemFieldDefinitions)) {
```
Comment (weisswurstkanone) [inline lib/Translation/ExportDataExtractorService/DataExtractor/DataObjectDataExtractor.php]: is the empty check necessary ?

File: lib/Translation/ExportDataExtractorService/DataExtractor/DataObjectDataExtractor.php [hunk 2]
```diff
@@ -151,6 +158,35 @@ protected function addLocalizedFields(DataObject\Concrete $object, AttributeSet
         return $this;
     }
 
+    /**
+     * @param Localizedfields $fd
+     * @param AttributeSet $result
+     * @param array|null $exportAttributes
+     */
+    protected function addBlocksInLocalizedfields(Localizedfields $fd, $definition, DataObject\Concrete $object, AttributeSet $result, array $exportAttributes = null)
```
Comment (weisswurstkanone) [inline lib/Translation/ExportDataExtractorService/DataExtractor/DataObjectDataExtractor.php]: please add $object to the phpdoc

**Comment by fashxp** [comment]
@mcuenez  do you have a change to look at the review comments?

**Comment by brusch** [comment]
@mcuenez any update? thanks!

**Comment by brusch** [comment]
@mcuenez last ping 😉

**Comment by elements-team** [comment]
Sorry, somehow I missed all the notifications :-/ Let me get back to you asap :-)

File: lib/Translation/ExportDataExtractorService/DataExtractor/DataObjectDataExtractor.php (hunk 1 above)
Comment (mcuenez) [reply lib/Translation/ExportDataExtractorService/DataExtractor/DataObjectDataExtractor.php]: Yes :-)

File: lib/Translation/ExportDataExtractorService/DataExtractor/DataObjectDataExtractor.php (hunk 2 above)
Comment (mcuenez) [reply lib/Translation/ExportDataExtractorService/DataExtractor/DataObjectDataExtractor.php]:
```suggestion
    /**
     * @param Localizedfields $fd
     * @param $definition
     * @param DataObject\Concrete $object
     * @param AttributeSet $result
     * @param array|null $exportAttributes
     */
    protected function addBlocksInLocalizedfields(Localizedfields $fd, $definition, DataObject\Concrete $object, AttributeSet $result, array $exportAttributes = null)
```

**Comment by mcuenez** [comment]
My apologies - i wasn't logged in with my account for a while :-)

**Comment by mcuenez** [comment]
> [CLA assistant check](https://cla-assistant.io/pimcore/pimcore?pullRequest=4529) 
> Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/pimcore/pimcore?pullRequest=4529) before we can accept your contribution.
> 
> **mcuenez** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).
> You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/pimcore/pimcore?pullRequest=4529) it.

I did sign the CLA, and verfied that my e-mail settings are correct. However, the `recheck` link does not update the status. Any suggestions?

**Comment by weisswurstkanone** [comment]
@mcuenez , ping regarding the CLA

**Comment by mcuenez** [comment]
@weisswurstkanone did you read my last comment? I don't know what else to do. Here's a screenshot of me signing the CLA:
image

If there is anything else I have to do in order to make this work, please, let me know.

**Pull request merged**